calico_cni/calico_cni.py

```diff
@@ -195,7 +196,7 @@ def add(self):
                   self.container_id)
 
         # Step 1: Assign an IP address using the given IPAM plugin.
```

Comment:
I think we need to do a little more re-structure for the ADD case.  Didn't spot this before.

For CNI, it is possible that the plugin will be called multiple times for multiple different CNI networks (all of which use Calico for networking).  In that case, we don't want to assign multiple IP addresses, we don't want to create another endpoint, and we don't want to set up another veth.  We just want to append a profile for the new network.

Let's talk about this before making the change.